src/nag-pack.ts

```diff
@@ -78,44 +96,38 @@ export interface IApplyRule {
   rule(node: CfnResource): NagRuleResult;
 }
 
-/**
- * The severity level of the rule.
- */
-export enum NagMessageLevel {
-  WARN = 'Warning',
-  ERROR = 'Error',
-}
-
 /**
  * Base class for all rule packs.
  */
 export abstract class NagPack implements IAspect {
-  protected verbose: boolean;
-  protected logIgnores: boolean;
-  protected reports: boolean;
-  protected reportStacks = new Array<string>();
+  protected loggers = new Array<INagLogger>();
   protected packName = '';
   protected userGlobalSuppressionIgnore?: INagSuppressionIgnore;
   protected packGlobalSuppressionIgnore?: INagSuppressionIgnore;
 
   constructor(props?: NagPackProps) {
-    this.verbose =
-      props == undefined || props.verbose == undefined ? false : props.verbose;
-    this.logIgnores =
-      props == undefined || props.logIgnores == undefined
-        ? false
-        : props.logIgnores;
-    this.reports =
-      props == undefined || props.reports == undefined ? true : props.reports;
     this.userGlobalSuppressionIgnore = props?.suppressionIgnoreCondition;
+    this.loggers.push(
+      new AnnotationLogger({
+        verbose: props?.verbose,
+        logIgnores: props?.logIgnores,
+      })
+    );
```

Comment:
Is it possible for a user disable the AnnotationLogger? Use case would be to fully customize the loggers that are used to handle logging/reporting outside of standard CDK synthesis annotation, which is blocking on error in the build.

As a CDK tools developer, I want to have full control over the reporting of CdkNag such as only making developers aware of issues while prototyping but not blocking the synthesis/deployment of code.

I realize this could be achieved by extending a given NagPack, and directly setting the `loggers` property in the constructor. Which maybe that is ideal, as it adds a bit of friction to go outside of the normal way of that CDKNag is expected to work.

Reply:
> it adds a bit of friction to go outside of the normal way of that CDKNag is expected to work.

That's the intent here. I don't think that the out of box `NagPack`s should allow for an opt out of this behavior considering the expected behavior of `cdk-nag` is the prevention of deployments.

In my view extending a `NagPack` and its props/ building your own `NagPack`  is the correct way to go about this since developers who want to do things like disable annotations likely would also want to be modifying/building their own rulesets and/or suppression conditions to the rules